arangod/Cluster/SynchronizeShard.cpp

```diff
@@ -171,10 +169,10 @@ static arangodb::Result getReadLockId(std::string const& endpoint,
       return arangodb::Result(TRI_ERROR_INTERNAL, error);
     }
   } else {
-    if (result) {
-      error.append(result->getHttpReturnMessage());
+    if (res.hasValue()) {
```

Comment:
technically this means there should be an exception in the Future

Reply:
@graetzer The if checks for a 200 response. Will there necessarily be an exception if it gets a 404 or something?

Reply:
no, just the Future contains either a value `network::Response` or an exception was thrown somewhere